src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -62,6 +64,274 @@ static ShardPlacement * TupleToShardPlacement(TupleDesc tupleDesc,
+{
+	DistTableCacheEntry *distTableCacheEntry = DistributedTableCacheEntry(relationId);
+	char *workerNodeName = workerNode->workerName;
+	uint32 workerNodePort = workerNode->workerPort;
+	List *shardIntervalList = NIL;
+	int shardIndex = 0;
+	int shardIntervalArrayLength = distTableCacheEntry->shardIntervalArrayLength;
+
+	for (shardIndex = 0; shardIndex < shardIntervalArrayLength; shardIndex++)
+	{
+		ShardPlacement *placementArray =
+			distTableCacheEntry->arrayOfPlacementArrays[shardIndex];
+		int numberOfPlacements =
+			distTableCacheEntry->arrayOfPlacementArrayLengths[shardIndex];
+		int placementIndex = 0;
+
+		for (placementIndex = 0; placementIndex < numberOfPlacements; placementIndex++)
+		{
+			ShardPlacement *placement = &placementArray[placementIndex];
+			char *shardNodeName = placement->nodeName;
+			uint32 shardNodePort = placement->nodePort;
+
+			if ((strcmp(shardNodeName, workerNodeName) == 0 &&
+				 shardNodePort == workerNodePort))
+			{
+				ShardInterval *shardInterval =
+					distTableCacheEntry->sortedShardIntervalArray[shardIndex];
+				shardIntervalList = lappend(shardIntervalList, shardInterval);
+			}
+		}
+	}
+
+	return shardIntervalList;
+}
+
+
+/*
+ * GenerateSizeQueryOnMultiplePlacements generates a select size query to get
+ * size of multiple tables from the relation with distributedRelationId. Note that,
+ * different size functions supported by PG are also supported by this function changing
+ * the size query given as the last parameter to function.
+ */
+StringInfo
+GenerateSizeQueryOnMultiplePlacements(Oid distributedRelationId, List *shardIntervalList,
+									  char *sizeQuery)
+{
+	/* schema name will be used to form shard qualified name */
+	Oid schemaId = get_rel_namespace(distributedRelationId);
+	char *schemaName = get_namespace_name(schemaId);
+
+	StringInfo selectQuery = makeStringInfo();
+	ListCell *shardCell = NULL;
+
+	appendStringInfo(selectQuery, "SELECT ");
+
+
+	foreach(shardCell, shardIntervalList)
+	{
+		ShardInterval *shardInterval = (ShardInterval *) lfirst(shardCell);
+		uint64 shardId = shardInterval->shardId;
+		char *shardName = get_rel_name(distributedRelationId);
+		char *shardQualifiedName = NULL;
+		char *quotedShardName = NULL;
+		AppendShardIdToName(&shardName, shardId);
+
+		shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
+		quotedShardName = quote_literal_cstr(shardQualifiedName);
+
+		appendStringInfo(selectQuery, sizeQuery, quotedShardName);
+		appendStringInfo(selectQuery, " + ");
+	}
+
+	/* there is no more size function so just add 0 instead of it */
+	appendStringInfo(selectQuery, "0;");
+	return selectQuery;
+}
+
+
+/*
+ * SupportFunctionForCitusSize is a helper function for all citus size functions. It
+ * first checks whether the table is distributed and size query can be run on it.
+ * Connection to each node has to be established to get the size of the table.
+ */
+uint64
+SupportFunctionForCitusSize(Oid relationId, char *sizeQuery)
```

Comment:
Note: When I read the name of this function I expected it to give a boolean answer on whether or not we "Support" something. To make it clear what this function does, maybe something like `GetDistributedTableSize`?

Also, shouldn't this function be static?